core/src/main/java/com/amplifyframework/core/AmplifyConfiguration.java

```diff
@@ -53,81 +57,115 @@
     private static final String DEFAULT_IDENTIFIER = "amplifyconfiguration";
 
     private final Map<String, CategoryConfiguration> categoryConfigurations;
+    private final Map<Platform, String> platformVersions;
```

Comment:
If order matters, you may want to use `LinkedHashMap` on the LHS of all uses. The Liskov principle is about using the type of least specificity. But for your use case, if ordering is essential, `Map` is not the type of least specificity.

Reply:
Wouldn't it automatically obtain an iterator from `LinkedHashMap` implementation everywhere I am referencing it because it is instantiated as one?